### ISSUE TITLE
[EPIC] IPv6 support

### ISSUE THREAD

**Issue by ApolloDS** [harvester/harvester#934]
**Is your feature request related to a problem? Please describe.**
When installing the ISO only IPv4 can be set, IPv6 is missing.

**Describe the solution you'd like**
IPv6 support

**Describe alternatives you've considered**
no alternatives.

**Additional context**
![grafik](https://user-images.githubusercontent.com/3502089/121159914-eaa95380-c84b-11eb-9807-1eafec34ea7f.png)

**Comment by telmich** [comment]
I'm also interested in giving harvester a try, but I only operate IPv6 only networks (i.e. not IPv4 at all)

**Comment by weihanglo** [comment]
Maybe related 

https://github.com/flannel-io/flannel/pull/1398
https://github.com/k3s-io/k3s/issues/284

**Comment by ronnyaa** [comment]
> I'm also interested in giving harvester a try, but I only operate IPv6 only networks (i.e. not IPv4 at all)

nothing greenfield have ipv4 for sure. shocked to see that harvester that  is fairly new is not ipv6 primarily.

**Comment by jhoelzel** [comment]
are there any updates here? Both flannel and k3s now support ipv6 dual-stack which should be perfect for most cases.

**Comment by jhoelzel** [comment]
are there any updates here? Both flannel and k3s now support ipv6 dual-stack which should be perfect for most cases.

**Comment by jhoelzel** [comment]
are there any updates here? Both flannel and k3s now support ipv6 dual-stack which should be perfect for most cases.

**Comment by bk201** [comment]
IPv6 is disabled at OS level in v1.1.1 due to k3s reports different node IP after upgrading: https://github.com/harvester/harvester/issues/2962

**Comment by itworksdoesntwork** [comment]
and how to enable it ?
we have plenty of ipv6 but as you guys can assume ipv4 is a little difficult to get and very expensive

**Comment by irony** [comment]
Any news here?

**Comment by joshuarestivo** [comment]
Bump.

**Comment by innobead** [comment]
cc @mingshuoqiu @starbops

**Comment by innobead** [comment]
@mingshuoqiu Please investigate this early, even though it's not planned for 1.7. We can implement it in phases, starting with IPv6 exclusive and then dual stack.

cc @harvester/network

**Comment by derhornspieler** [comment]
Any updates for ipv6 to support dual stack capabilities?  IPv6 is sort of critical in today's infrastructure designs and commonly used.  Hopefully this was not dropped from v1.7.0 and maybe even an option to enable in v1.6.0 as beta?

**Comment by agowa** [comment]
All I found so far for getting IPv6 to work in harvester is to manually patch it into everything bypassing basically harvester entirely...

https://harvester.github.io/tests/manual/upgrade/ipv6-dhcp-upgrade/

**Comment by innobead** [comment]
@martindekov @starbops I assume we will create subtasks for phased implementation across the following releases. Moved this EPIC to `planning` first, and we should add the subtasks planned for 1.9.0 to the 1.9.0 milestone. 

cc @harvester/tpm @harvester/network

**Comment by martindekov** [comment]
Yes I wanted to gather to agree on the MVP and then once we fully see the scope we follow with design and implementation and then figure out how issues should be made and in which order so the core is delivered sequentially (defining the APIs + base host network set up) and then in parallel we can unlock delivering the other components on top (mgmt/cluster/vm network/kube-ovn/kube-vip/ui/doc) while not blocking other development.

**Comment by rrajendran17** [comment]
IMO, we can create subtasks based on following networking functionalities/areas in Harvester and order of priority spanning different phases.
1.MGMT network physical connectivity(node IP/VIP)
2.Cluster/Pod Networking
3. VM Networking
4. Guest clusters
5. VM Migration Network,Storage Network
6. KubeOVN Overlay Networking
7. HostNetworkConfig

**Comment by martindekov** [comment]
At the moment I hit this:
https://github.com/rancher/rancher/issues/50933

If I work around it by forcing ipv4 single stack service through manual patch then I hit another follow up problem:
```
2026/05/04 07:58:23 [INFO] [planner] rkecluster fleet-local/local: waiting for at least one control plane, etcd, and worker node to be registered
```

And the machine plan won't be populated:
```
node-1:/home/rancher # kubectl get secrets -A | grep custom
fleet-local                              custom-d4a817aff8bc-machine-bootstrap                               rke.cattle.io/bootstrap                       1      6m31s
fleet-local                              custom-d4a817aff8bc-machine-bootstrap-token-rzxmw                   kubernetes.io/service-account-token           3      6m31s
fleet-local                              custom-d4a817aff8bc-machine-plan                                    rke.cattle.io/machine-plan                    0      6m31s
fleet-local                              custom-d4a817aff8bc-machine-plan-token-77m74                        kubernetes.io/service-account-token           3      6m31s
```

I think this might be a blocker for the feature for the ipv6 first dual stack path.

**Comment by martindekov** [comment]
Moved the issue to icebox until we decide if we will move forward with ipv4 + ipv6 dual stack current state:
* ipv4 🟢 (currently supported)
* dual stack ipv4 + ipv6 - 🟢 [POC Code](https://github.com/martindekov/harvester-installer/tree/ipv6-hep-poc-dual-stack-ipv4-first)
* dual stack ipv6 + ipv4 - 🔴 [Bug (Blocker)](https://github.com/rancher/rancher/issues/50933)
* ipv6 - ⚪ - hardest to POC/testing infra

**Comment by agowa** [comment]
> ipv6 - ⚪ - hardest to POC/testing infra

What exactly do you mean by this? What are the problems that appear to make it hard here?

**Comment by ronnyaa** [comment]
ipv6-only is also the most important to get right. since that is the goal of all of our migrations. dualstack is extra complexity, for very little extra gain.

**Comment by agowa** [comment]
Jup, also it is way easier to implement dualstack when you've ipv6-only first, as you can just NAT46/NAT64 on the ingress (and if really required also right before the workload). No need to do it anywhere in between.

**Comment by martindekov** [comment]
Hey @agowa this just reports POC progress. IPV6 only is hardest to POC as we did a lot of work and assume ipv4 in a a lot more places.

**Comment by forbesguthrie** [comment]
Updated title to make it more obvious this is to track IPv6-only, as there's a separate epic for the IPv6 Dual Stack work.
